### PR TITLE
layout: Have `current_block_direction_position` in `PlacementState` use `Au`

### PR DESCRIPTION
**Pull request by ektuu**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes are part of #29819.
- [x] These changes do not require tests because it's a type adjustment and doesn't alter the underlying logic

### CODE REVIEW

**Comment by mrobinson** [comment]
>  [x] These changes enable current_block_direction_position to accept Au

Instead of modifying this line of the template to say what the change does, your description should go into the body of the PR description itself (so it is included with the final commit in the repository). This line of the template is to remind you to reference open bugs if they exist. Note that prefixing the reference with "fix" or similar will automatically close the bug when a change is merged (see https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

**Review by mrobinson — APPROVED** [review]

**Comment by ektuu** [comment]
> > [x] These changes enable current_block_direction_position to accept Au
> 
> Instead of modifying this line of the template to say what the change does, that description should go into the body of the PR message itself (so it is included with the final commit in the repository). This line of the template is to remind you to reference open bugs if they exist. Note that prefixing the reference with "fix" or similar will automatically close the bug when a change is merged (see https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

Got it!

File: components/layout_2020/flow/mod.rs [hunk 1]
```diff
@@ -1621,24 +1621,24 @@ impl PlacementState {
                 }
                 fragment.content_rect.start_corner.block +=
                     Length::from(self.current_margin.solve()) +
-                        self.current_block_direction_position;
+                        self.current_block_direction_position.into();
```
Comment (Loirooriol) [inline components/layout_2020/flow/mod.rs:1624]: Nit: you can do the sum before converting:

```rs
                fragment.content_rect.start_corner.block +=
                    (self.current_margin.solve() + self.current_block_direction_position).into();
```

File: components/layout_2020/flow/mod.rs [hunk 2]
```diff
@@ -1621,24 +1621,24 @@ impl PlacementState {
...
-                        Length::from(self.current_margin.solve()) + fragment_block_size.into();
+                        self.current_margin.solve() + fragment_block_size;
                     self.current_margin = fragment_block_margins.end;
                 }
             },
             Fragment::AbsoluteOrFixedPositioned(fragment) => {
                 let offset = LogicalVec2 {
                     block: Length::from(self.current_margin.solve()) +
-                        self.current_block_direction_position,
+                        self.current_block_direction_position.into(),
```
Comment (Loirooriol) [inline components/layout_2020/flow/mod.rs:1641]: Ditto:

```rs
                    block: (self.current_margin.solve() + self.current_block_direction_position)
                        .into(),
```

File: components/layout_2020/flow/mod.rs (hunk 1 above)
Comment (ektuu) [reply components/layout_2020/flow/mod.rs]: Sure. I will apply the changes.

**Review by Loirooriol — APPROVED** [review]

**Comment by Loirooriol** [comment]
There was a failure, not sure if it was an intermittent or this patch is changing the result by a tiny amount.
https://github.com/servo/servo/actions/runs/8490170798/job/23261567690

**Comment by ektuu** [comment]
> There was a failure, not sure if it was an intermittent or this patch is changing the result by a tiny amount. https://github.com/servo/servo/actions/runs/8490170798/job/23261567690


Could you guide me on what to do next about the test failure?

**Comment by Loirooriol** [comment]
I think it's fine to update the expectation for now, to expect a failure. Hopefully it will pass again when switching flexbox into app units.

**Comment by ektuu** [comment]
> I think it's fine to update the expectation for now, to expect a failure. Hopefully it will pass again when switching flexbox into app units.

Okay.
How would you feel about me working on updating the flexbox implementation to use app units (Au)? Do you think that's a good next step for me?

**Comment by ektuu** [comment]
Hello @mrobinson @Loirooriol,

I hope your day is going well. I'm interested in continuing my contributions to Servo. Could you please direct me to any open issues or provide guidance?

**Comment by Loirooriol** [comment]
@ektuu Why not first finish this PR? It needs to update test expectations.

**Comment by ektuu** [comment]
> @ektuu Why not first finish this PR? It needs to update test expectations.

Certainly.
Based on your earlier recommendation, I'll transition the flexbox implementation to utilize app units.

**Comment by mrobinson** [comment]
I pushed a new version of this with the test expectation.

**Pull request merged**